server/events/working_dir.go

```diff
@@ -307,6 +309,20 @@ func (w *FileWorkspace) cloneDir(r models.Repo, p models.PullRequest, workspace
 	return filepath.Join(w.repoPullDir(r, p), workspace)
 }
 
+func (w *FileWorkspace) IsFileTracked(log logging.SimpleLogging, cloneDir string, filename string) (bool, error) {
+	cmd := exec.Command("git", "ls-files", filename)
+	cmd.Dir = cloneDir
+
+	output, err := cmd.CombinedOutput()
+
+	if err != nil {
+		log.Warn("Error checking if %s is tracked: %s", filename, string(output))
+		return false, err
+	}
+	return len(output) > 0, nil
```

Comment:
I'm more in favor with just keeping all this logic in `core/runtime/common` or even in `InitStepRunner` 
1) I dont know if there is really going to be something else that uses this logic. The long term solution is to just clean up after each operation
2). `ProjectCommandContext` is becoming a dumping ground for things we need to plumb through the service and is losing it's meaning, let's not add to that unnecessarily.
3) We can leave `ProjectCommandRunner` out of this, which already has a lot going on.

Reply:
I have made the suggested change and moved the code to check if file is tracked in git into `core/runtime/common`